### DISCUSSION TITLE
'Enter','ArrowRight','ArrowLeft' as Invalid key string

### DISCUSSION THREAD

**Discussion by BastienBlain** [jspsych/jsPsych#1559] [Q&A] [answered]
Hi there,

I'm using jspsych-6.2.0 and I specified the following plugin for example as below: 

```
var XX = {
  type: "html-keyboard-response",
  stimulus: "YY",
  choices: 'ArrowRight',
};
```

Nothing happened when I did  then press the right arrow while seeing YY, that is, the next screen is not displayed. There is no error message. I tried chrome and internet explorer and both lead to the same issue. 

If instead I replace 'ArrowRight' by 'f', then pressing 'f' works (in that I move to the next screen). 

Pressing the right arrow e.g. on https://www.freecodecamp.org/news/javascript-keycode-list-keypress-event-key-codes/ does return the relevant code.

Any idea of what could be wrong? 

Thanks for your help!

**Comment by becky-gilbert** [comment]
Converting this to a discussion.

**Comment by becky-gilbert** [comment, marked as answer]
Hi @BastienBlain! Try this:
```js
var XX = {
  type: "html-keyboard-response",
  stimulus: "YY",
  choices: ['rightarrow']
};
```
Long answer: I think you're running into two different problems here.
1. The `choices` parameter must be an _array_, e.g. `choices: ['f']`. If you use a string, then jsPsych will treat the string as an array, which means that every character in the string will become a separate valid key choice. This means that using `choices: 'f'` (without the square brackets) will actually work as expected, even though 'f' isn't inside an array. But if you use `choices: 'ArrowRight'`, this string will become an array: 'a','r','r','o'... etc. You can see this for yourself by using `choices: 'ArrowRight'` - you'll find that the individual letters become valid keys. (Side note: ideally jsPsych would check that all parameters are using the correct types to avoid this problem - this is something we should fix in the future)
2. The 'ArrowRight' key identifier won't work because that's actually the `key` event's `.code` value, whereas jsPsych v6.2 uses the numeric `.keyCode` values (I know, very confusing!). So you could try using the numeric keyCode for the right arrow key, which is 39. Or you can use 'rightarrow', which is the key string identifier that jsPsych v6.2 uses. You can find this information in the `keylookup` object in `jspsych.js` here: https://github.com/jspsych/jsPsych/blob/42962039af40f1ec6dfc1dadc65b4c9832dcd43e/jspsych.js#L2204-L2298

**Reply by BastienBlain** [reply]
Thank you very much indeed for your very quick and thorough answer! I now understand this much better!

**Reply by BastienBlain** [reply]
May I ask the same question but with JsPsych 7.1? Left and right arrows seem not to be recognised if I specified them as before: 
`  choices: ['leftarrow','rightarrow']`
The mapping is no longer in jspsych.js or not in a trivial way. 

Thanks!

**Reply by BastienBlain** [reply]
I guess it is now more standard, like this? 
`choices: ['ArrowLeft','ArrowRight']`

**Reply by jodeleeuw** [reply]
If you use [this handy tool](https://keycode.info/) it is whatever is shown under `event.key`
